Assets/00_Content/Scripts/Player/PlayerPickUp.cs

```diff
@@ -0,0 +1,93 @@
+using Interactables;
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using System.Linq;
+
+namespace Player {
+
+	public class PlayerPickUp : MonoBehaviour {
+
+		[SerializeField] private Transform playerGrabTransform;
+
+		//Should probably be displayed on a canvas instead.
+		[SerializeField] private GameObject highlight;
+
+		private List<PickUp> pickUps;
+		private PickUp closestPickUp;
+		private PickUp pickedUpItem;
+		private SphereCollider sphereCollider;
+
+		private void Start() {
+
+			pickUps = new List<PickUp>();
+			sphereCollider = GetComponent<SphereCollider>();
+		}
+
+		public void PickUpClosestItem() {
+			HighlightPickUp(null);
+
+			if (pickUps.Count < 1 || closestPickUp == null)
+				return;
+
+			closestPickUp.PickUpItem(playerGrabTransform);
+			pickedUpItem = closestPickUp;
+			pickUps.Clear();
+			sphereCollider.enabled = false;
+		}
+
+		public void DropItem() {
+
+			if (pickedUpItem == null)
+				return;
+
+			pickedUpItem.DropItem();
+			pickedUpItem = null;
+			sphereCollider.enabled = true;
+		}
+
+		public void UseItem() {
+			if (pickedUpItem != null && pickedUpItem is IUseable)
+				Debug.Log("Using item...");
+		}
+
+		private void OnTriggerEnter(Collider other) {
+
+			if (other.gameObject.CompareTag("PickUp") && pickedUpItem == null) {
+
+				pickUps.Add(other.gameObject.GetComponent<PickUp>());
+				closestPickUp = pickUps.OrderBy(x => (x.transform.position - transform.position).sqrMagnitude).First();
```

Comment:
Performing this calculation on enter/exit can lead to unintuitive behaviour. As long as no new pickupable enter or exit it will always be the same that is "closest" regardless of player position. To keep the highlight working would this not need to be done in for example Update?